src/software/ai/navigator/navigator.cpp

```
@@ -44,56 +48,57 @@ void Navigator::visit(const KickIntent &kick_intent)
-                        Util::DynamicParameters::Navigator::transition_speed_factor
-                            .value());
+        current_destination = (path_points)[1];
+        double segment_final_vel =
+            getCloseToEnemyObstacleFactor((path_points)[1]) *
+            calculateTransitionSpeedBetweenSegments(
+                (path_points)[0], (path_points)[1], (path_points)[2],
+                ROBOT_MAX_SPEED_METERS_PER_SECOND *
+                    Util::DynamicParameters::Navigator::transition_speed_factor
+                        .value());
 
-            auto move = std::make_unique<MovePrimitive>(
-                move_intent.getRobotId(), current_destination,
-                move_intent.getFinalAngle(), segment_final_vel,
-                move_intent.isDribblerEnabled(), move_intent.isSlowEnabled(),
-                move_intent.getAutoKickType());
-            current_primitive = std::move(move);
-            return;
-        }
-        if ((*path_points).size() == 2)
-        {
-            current_destination = (*path_points)[1];
-            auto move           = std::make_unique<MovePrimitive>(
-                move_intent.getRobotId(), current_destination,
-                move_intent.getFinalAngle(),
-                move_intent.getFinalSpeed() *
-                    getCloseToEnemyObstacleFactor((*path_points)[1]),
-                move_intent.isDribblerEnabled(), move_intent.isSlowEnabled(),
-                move_intent.getAutoKickType());
-            current_primitive = std::move(move);
-            return;
-        }
+        auto move = std::make_unique<MovePrimitive>(
+            move_intent.getRobotId(), current_destination,
+            move_intent.getFinalAngle(), segment_final_vel,
+            move_intent.isDribblerEnabled(), move_intent.isSlowEnabled(),
+            move_intent.getAutoKickType());
+        current_primitive = std::move(move);
+    }
+    else if ((path_points).size() == 2)
```

there is a check for > 2 and == 2 for path_point size, should we raise an exception if size < 2

---

good point, we use to return stop but I moved things around too much